medicines/doc-index-updater/src/storage_client/azure_blob_client.rs

```diff
@@ -55,17 +55,16 @@ impl AzureBlobStorage {
         match base64::decode(&self.master_key) {
             Ok(_) => Ok(
                 Client::new(&self.storage_account, &self.master_key).map_err(|e| {
-                    StorageClientError::ClientError {
-                        message: format!("Couldn't create storage client: {:?}", e),
-                    }
+                    StorageClientError::ClientError(format!(
+                        "Couldn't create storage client: {:?}",
+                        e
+                    ))
```

Comment:
Think this `Ok` condition code would read more clearly (reduced nesting) if it was extracted out to a short function that takes care of creating the client or returning the mapped error.